src/txn/txn_ckpt.c

```diff
@@ -921,6 +1017,14 @@ __checkpoint_worker(
 			WT_ERR(bm->checkpoint_resolve(bm, session));
 	}
 
+	/*
+	 * Update the checkpoint generation for this handle so visible
+	 * updates newer than the checkpoint can be evicted.
+	 */
+	btree->checkpoint_gen = conn->txn_global.checkpoint_gen;
+	WT_STAT_FAST_DATA_SET(session,
+	    btree_checkpoint_generation, btree->checkpoint_gen);
+
```

Comment:
Did you consider moving this into the `done:` block so we bump the `checkpoint_gen` of clean files?

I think it would only ever be a marginal benefit, but if the checkpoint took long enough, it might be good to be able to evict from files that become dirty after a checkpoint starts?